nodes.go

```diff
@@ -161,7 +161,7 @@ func (nodes Nodes) FlattenAll(document *Document, result Nodes) {
 func (nodes Nodes) deleteNode(n Node) (Nodes, bool) {
 	for i, node2 := range nodes {
 		if node2 == n {
-			return append(nodes[:i], nodes[i+2:]...), true
+			return append(nodes[:i], nodes[i+1:]...), true
```

Comment:
Can you please explain why it is removing two nodes here?

Reply:
I'm not sure what you mean?  This code _should_ only be removing one node;  without this change I was routinely getting out of bounds errors.

```
package main

import (
	"fmt"
	"github.com/elliotchance/gedcom"
)

func main() {
	d := gedcom.NewDocument()
	i := d.AddIndividual("p1")
	i.AddName("Test /Person/")
	i2 := d.AddIndividual("p2")
	i2.AddName("Test /Maiden/")
	f := d.AddFamilyWithHusbandAndWife("f1", i, i2)

	f.SetWife(nil)
	fmt.Println(d.GEDCOMString(0))
	
}
```
`panic: runtime error: slice bounds out of range [3:2]`

So I changed it to only slice out one node.

Reply:
I'm sorry, I was reading the wrong direction. It was a bug in the original code - thanks for fixing.